x-pack/plugins/aiops/public/hooks/use_overall_stats.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useEffect, useState, useMemo } from 'react';
+import { lastValueFrom } from 'rxjs';
+import { i18n } from '@kbn/i18n';
+import type { ToastsStart } from '@kbn/core/public';
+import { useAiOpsKibana } from '../kibana_context';
+import { extractErrorProperties } from '../../common/error_utils';
+import {
+  DocumentCountStats,
+  getDocumentCountStatsRequest,
+  processDocumentCountStats,
+  OverallStatsSearchStrategyParams,
+} from '../get_document_stats';
+
+export interface OverallStats {
+  totalCount: number;
+  documentCountStats?: DocumentCountStats;
+}
+
+function displayError(toastNotifications: ToastsStart, index: string, err: any) {
+  if (err.statusCode === 500) {
+    toastNotifications.addError(err, {
+      title: i18n.translate('xpack.aiops.index.dataLoader.internalServerErrorMessage', {
+        defaultMessage:
+          'Error loading data in index {index}. {message}. ' +
+          'The request may have timed out. Try using a smaller sample size or narrowing the time range.',
+        values: {
+          index,
+          message: err.error ?? err.message,
+        },
+      }),
+    });
+  } else {
+    toastNotifications.addError(err, {
+      title: i18n.translate('xpack.aiops.index.errorLoadingDataMessage', {
+        defaultMessage: 'Error loading data in index {index}. {message}.',
+        values: {
+          index,
+          message: err.error ?? err.message,
+        },
+      }),
+    });
+  }
+}
+
+export function useOverallStats<TParams extends OverallStatsSearchStrategyParams>(
```

Comment:
`useOverallStats` doesn't make sense here in this context since you only need the document count. Suggestion to rename this file, this hook, and `setOverallStats`.